src/scipp/testing/strategies.py

```diff
@@ -0,0 +1,225 @@
+    if sizes is None:
+        sizes = sizes_dicts(ndim)
+    if isinstance(sizes, st.SearchStrategy):
+        sizes = draw(sizes)
+
+    if unit is None:
+        unit = units()
+    if isinstance(unit, st.SearchStrategy):
+        unit = draw(unit)
+
+    if dtype is None:
+        # TODO other dtypes?
+        dtype = scalar_numeric_dtypes()
+    if isinstance(dtype, st.SearchStrategy):
+        dtype = draw(dtype)
+
+    if with_variances is None:
+        with_variances = use_variances(dtype)
+    if isinstance(with_variances, st.SearchStrategy):
+        with_variances = draw(with_variances)
+
+    return dict(sizes=sizes,
+                unit=unit,
+                dtype=dtype,
+                with_variances=with_variances,
+                elements=elements,
+                fill=fill,
+                unique=unique)
+
+
+def variables(*,
+              ndim=None,
+              sizes=None,
+              unit=None,
+              dtype=None,
+              with_variances=None,
+              elements=None,
+              fill=None,
+              unique=None) -> st.SearchStrategy:
+    args = variable_args(ndim=ndim,
+                         sizes=sizes,
+                         unit=unit,
+                         dtype=dtype,
+                         with_variances=with_variances,
+                         elements=elements,
+                         fill=fill,
+                         unique=unique)
+    return args.flatmap(lambda a: _variables_from_fixed_args(a))
+
+
+def n_variables(n: int,
+                *,
+                ndim=None,
+                sizes=None,
+                unit=None,
+                dtype=None,
+                with_variances=None,
+                elements=None,
+                fill=None,
+                unique=None) -> st.SearchStrategy:
+    args = variable_args(ndim=ndim,
+                         sizes=sizes,
+                         unit=unit,
+                         dtype=dtype,
+                         with_variances=with_variances,
+                         elements=elements,
+                         fill=fill,
+                         unique=unique)
+    return args.flatmap(
+        lambda a: st.tuples(*(_variables_from_fixed_args(a) for _ in range(n))))
+
+
+@st.composite
+def _make_vectors(draw, sizes):
+    values = draw(npst.arrays(float, (*sizes.values(), 3)))
+    return creation.vectors(dims=tuple(sizes), values=values, unit=draw(units()))
+
+
+@st.composite
+def vectors(draw, ndim=None) -> st.SearchStrategy:
+    if ndim is None:
+        ndim = draw(st.integers(0, 3))
+    return draw(sizes_dicts(ndim).flatmap(lambda s: _make_vectors(s)))
+
+
+@st.composite
+def coord_dicts_1d(draw, *, coords, sizes, args=None) -> dict:
+    args = (args or {})
+    args['sizes'] = sizes
+    try:
+        del args['ndim']
+    except KeyError:
+        pass
+    return {dim: draw(variables(**args)) for dim in coords}
+
+
+class _NotSetType:
```

Comment:
Not sure I understand why you need this. Could you not just use `coords=None` ?

Reply:
This is used to generate no coords. But `NotSet` generates dim-coords.

Reply:
But we could use `[]` to mean 'no coords'